### PR TITLE
Typescript types and use of peerDependencies

### PR DESCRIPTION
**Pull request by mharj**
- Typescript type definitions.
- dev package updates.
- move dependencies to peerDependencies (add current appInsight and winston only major version lockdown).

Tested on current TS based project with npm link and seem to be happy with it.

### CODE REVIEW

File: lib/winston-azure-application-insights.d.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+declare module "winston-azure-application-insights" {
+  import type * as appInsights from "applicationinsights";
+  import type * as Transport from "winston-transport";
```
Comment (cherealnice) [inline lib/winston-azure-application-insights.d.ts]: Should just be `import Transport from "winston-transport";`

Comment (mharj) [reply lib/winston-azure-application-insights.d.ts]: seems to be ok TS both ways, but on module it's just "export = " so changed to use directly.

Comment (cherealnice) [reply lib/winston-azure-application-insights.d.ts]: It yelled at me when trying to import the module into another typescript file. Did you not have that problem?

**Review by willmorgan — APPROVED** [review]
LGTM - @dhensby are you still using this lib, and able to test it out?

**Comment by dhensby** [comment]
I don't, I'm afraid :(

**Comment by orgads** [comment]
When is the next release planned? We copied the declarations file locally, but will love to use it from the package.

**Pull request merged**